Utils/ReadDynamicSizeArray.java

```diff
@@ -0,0 +1,30 @@
+package Utils;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class ReadDynamicSizeArray {
+    private int numberOfLinesInArray;
+
+    private int numberOfElementsPerLine;
+
+    private int element;
+
+    public ArrayList[] readDynamicArray() {
+        Scanner scanner = new Scanner(System.in);
+        ArrayList[] arrayList = new ArrayList[20000];
```

Comment:
That's nice that you're using ArrayList, but you can make the solution much more elegant if you use generics.